tests/test_analyses.py

```diff
@@ -37,3 +37,83 @@ def test_roc(client):
     for key in keys:
         for value in resp.json[key]:
             assert value in {'Inf', '-Inf'} or 0 <= value <= 1
+
+
+def test_plsda(client):
+    for i in range(2):  # run test on two datasets
```

Comment:
You might parametrize this test rather than doing both together.  https://docs.pytest.org/en/stable/parametrize.html#parametrize-basics